pkg/virtctl/create/instancetype/instancetype_test.go

```diff
@@ -0,0 +1,184 @@
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "1"),
+				setFlag(MemoryFlag, "128Mi"),
+				setFlag(GPUFlag, "name:gpu1,devicename:nvidia"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err = getInstancetypeSpec(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(instancetypeSpec.GPUs).To(HaveLen(1))
+			Expect(instancetypeSpec.GPUs[0].Name).To(Equal("gpu1"))
+			Expect(instancetypeSpec.GPUs[0].DeviceName).To(Equal("nvidia"))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("should succeed with defined hostDevices", func(namespaced string) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "1"),
+				setFlag(MemoryFlag, "128Mi"),
+				setFlag(HostDeviceFlag, "name:device1,devicename:intel"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err = getInstancetypeSpec(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(instancetypeSpec.HostDevices).To(HaveLen(1))
+			Expect(instancetypeSpec.HostDevices[0].Name).To(Equal("device1"))
+			Expect(instancetypeSpec.HostDevices[0].DeviceName).To(Equal("intel"))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("should succeed with valid IOThreadsPolicy", func(namespaced, param string, policy v1.IOThreadsPolicy) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "1"),
+				setFlag(MemoryFlag, "128Mi"),
+				setFlag(IOThreadsPolicyFlag, param),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err := getInstancetypeSpec(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*instancetypeSpec.IOThreadsPolicy).To(Equal(policy))
+
+		},
+			Entry("VirtualMachineInstacetype set to auto", namespaced, "auto", v1.IOThreadsPolicyAuto),
+			Entry("VirtualMachineInstacetype set to shared", namespaced, "shared", v1.IOThreadsPolicyShared),
+
+			Entry("VirtualMachineClusterInstacetype set to auto", "", "auto", v1.IOThreadsPolicyAuto),
+			Entry("VirtualMachineClusterInstacetype set to shared", "", "shared", v1.IOThreadsPolicyShared),
+		)
+
+		DescribeTable("invalid cpu and memory", func(cpu, memory, errMsg string) {
+			err := clientcmd.NewRepeatableVirtctlCommand(create, Instancetype, namespaced,
+				setFlag(CPUFlag, cpu),
+				setFlag(MemoryFlag, memory),
+			)()
+
+			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError(errMsg))
+		},
+			Entry("Ivalid cpu string value", "two", "256Mi", "invalid argument \"two\" for \"--cpu\" flag: strconv.ParseUint: parsing \"two\": invalid syntax"),
+			Entry("Ivalid cpu negative value", "-2", "256Mi", "invalid argument \"-2\" for \"--cpu\" flag: strconv.ParseUint: parsing \"-2\": invalid syntax"),
```

Comment:
Just noticed we still need additional validation because 0 is a valid uint32 value but not valid in the context of CPU count.